library/std/src/sys/pal/windows/process.rs

```diff
@@ -337,18 +334,15 @@ impl Command {
 
         let si_ptr: *mut c::STARTUPINFOW;
 
-        let mut proc_thread_attribute_list;
         let mut si_ex;
 
-        if !self.proc_thread_attributes.is_empty() {
+        if let Some(proc_thread_attribute_list) = proc_thread_attribute_list {
             si.cb = mem::size_of::<c::STARTUPINFOEXW>() as u32;
             flags |= c::EXTENDED_STARTUPINFO_PRESENT;
 
-            proc_thread_attribute_list =
-                make_proc_thread_attribute_list(&self.proc_thread_attributes)?;
             si_ex = c::STARTUPINFOEXW {
                 StartupInfo: si,
-                lpAttributeList: proc_thread_attribute_list.0.as_mut_ptr() as _,
+                lpAttributeList: proc_thread_attribute_list.as_mut_ptr().cast::<c_void>(),
```

Comment:
I don't really understand why the win API require a `*mut` to the attribute list here.

Reply:
Because the C headers do not distinguish. between const and mut so, when automatically generating the bindings, `mut` is the most reasonable option. You can open an issue on the [win32metadata repo](https://github.com/microsoft/win32metadata) if you'd like them to manually add some metadata so the Rust bindings know which one to use.

Though it doesn't matter too much. In rust, const and mut on pointers is just a lint. You can do, e.g. `proc_thread_attribute_list.as_ptr().cast::<c_void>().cast_mut()` to go from a reference to a pointer.

Reply:
I thought that was highly illegal because of some `llvm` attributes that rust sets when dealing with shared refenrences.

Reply:
Pointers and references are different. You cannot turn a `&` into an `&mut`. You can turn an `&` into a `*mut` so long as you don't ever write to the memory.

Reply:
Yeah, my concern was that with the win api having an interface that explicitly expects a `*mut` I thought it might be the case that they actually write to it, but good to know. I will add a safety comment and open an issue over at the win32 crate repo.